File: poo/python/Carro/Carro.py
#!/usr/bin/python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 27/03/14
"""

class Carro:

    # Atributos #
    __acelerou = None
    __freou = None
    
    # Construtor #
    def __init__(self, acelerou=0, freou=0):
        self.__acelerou = acelerou
        self.__freou = freou

    # Getters #
    def getAcelerou(self):
        return self.__acelerou
    def getFreou(self):
        return self.__freou

    # Setters #
    def setAcelerou(self, acelerou):
        self.__acelerou = acelerou
    def setFreou(self, freou):
        self.__freou = freou

    # Acelerar o carro #
    def acelerar(self, a):
       self.__acelerou += 1
       print('\tAcelerando o carro para a velocidade %s' % (a))
    
    # Frear o carro #
    def frear(self):
        self.__freou += 1
        print('\tFreando o carro')

    # Quantidade de freadas #
    def vezesFreou(self):
        print('\tFreou %d vezes' % (self.__freou))

    # Quantidade de aceleradas #
    def vezesAcelerou(self):
        print('\tAcelerou %d vezes' % (self.__acelerou))


if __name__ == '__main__':

    c1 = Carro()
    c2 = Carro()

    print('Objeto c1: ')
    c1.acelerar(10)
    c1.frear()
    c1.vezesFreou()
    c1.vezesAcelerou()
    
    print('Objeto c2: ')
    c2.acelerar(15)
    c2.frear()
    c2.vezesFreou()
    c2.vezesAcelerou()

File: poo/c++/Carro/Carro.cpp
#include <iostream>
#include <string>

class Carro{

    /* Atributos */
    private:
        int acelerou;
        int freou;
        int id;

    /* Construtor */
    public:

        Carro(int a){
            acelerou = 0;
            freou = 0;
            id = a;
        }

        ~Carro(){
            std::cout << "Chamando o destrutor" << "\n";
        }

        /* Getters */
        int getAcelerou(){
            return acelerou;
        }

        int getFreou(){
            return freou;
        }

        /* Setters */
        void setAcelerou(int acelerou){
            acelerou = acelerou;
        }

        void setFreou(int freou){
            freou = freou;
        }

        void acelerar(int a){
            setAcelerou(getAcelerou()+1);
            std::cout << "\tAcelerando o carro para: " << a << "\n";
        }

        void frear(){
            setFreou(getFreou()+1);
            std::cout << "\tFreando o carro" << "\n";
        }

        void vezesFreou(){
            std::cout << "\tFreou " << getFreou() << " vezes" << "\n";
        }

        void vezesAcelerou(){
            std::cout << "\tAcelerou " << getAcelerou() << " vezes" << "\n";
        }

};

int main(){

    Carro c1 (1);
    Carro c2 (2);

    std::cout << "Objeto c1: " << "\n";
    c1.vezesFreou();
    c1.vezesAcelerou();

    std::cout << "Objeto c2: " << "\n";
    c2.acelerar(15);
    c2.frear();
    c2.vezesFreou();
    c2.vezesAcelerou();

    return 0;
}
